src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { ReactElement, ReactNode } from "react";
+import styled, { css } from "styled-components";
+import { Menu } from "@headlessui/react";
+import { IconDirection } from "../Icon/Direction";
+import { Text } from "../../typography";
+import { Button } from "../Button";
+import Icon from "../Icon";
+
+export interface DropdownOption {
+  value: string;
+  label: string;
+}
+
+export interface DropdownProps {
+  buttonContent: ReactNode;
+  options: DropdownOption[];
+  selected?: DropdownOption;
+  setSelected: (option: DropdownOption) => void;
+}
+
+export const Dropdown = ({
+  buttonContent,
+  options,
+  selected,
+  setSelected,
+}: DropdownProps): ReactElement => {
+  const isDropdownOptionSelected = ({ value }: DropdownOption): boolean => {
+    return value === selected?.value;
+  };
+
+  return (
+    <Menu>
+      {({ open }) => (
+        <>
+          <StyledMenuButton>
+            {buttonContent}
+            <ChevronComponent open={open} />
+          </StyledMenuButton>
+          <StyledMenuItems>
+            {options.map((option) => (
+              <Menu.Item key={option.value} onClick={() => setSelected(option)}>
+                {({ active }) => (
+                  <StyledMenuItem
+                    active={active || isDropdownOptionSelected(option)}
+                  >
+                    {option.label}
```

Comment:
@Major-Eth when we select an option, do we want the button label to be replaced? It seems a bit unclear, and if so, should we have a default value when we load the dropdown menu?

We can also leave that for when we have more clarity how the dropdown menu will be used.

Reply:
Yes, the label should change to be the selected option. We should maybe provide a defaultIndex/defaultKey to select which one should be selected on load ?

Reply:
Sounds good, so I was thinking of adding an optional prop `defaultValue` as https://react-select.com/props#statemanager-props is doing to the `DropdownProps`

The problem now is the icon, I've allowed the button to have an icon, but not the dropdown options. Do we want to support the options to have an icon? If so, I think we could make the `label` not a string but a `ReactNode` and let the user of the component put there what they wish. What are your thoughts?

Reply:
@Major-Eth added support for the default option with https://github.com/Major-Eth/yearn-web-lib/pull/28/commits/a3d688c30efe648f27a7dabe59e80d814f98d9ac

Now we need to think if we want to support icons in the dropdown options. Perhaps something along the lines

```typescript
interface DropdownOption {
  value: string;
  label: string;
  icon: ReactElement;
}
```

It doesn't appear to be a requirement in the issue, but I did see in the mocks that we are using an icon in a dropdown menu. We can always come back to this later if we need to support icons. Let me know.

Fibo has approved, do you think this is good to merge?